### PR TITLE
Add docs for m2

### PR DESCRIPTION
**Pull request by Scones1016**
@nvasilakis @GNtousakis @vagos @nikpag

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+**Node configuration.** At a minimum, the node configuration includes an IP address and a port. In the examples below, we assume two nodes — by default a node `n1`, which at times initiates interactions with a node `n2`. 
+
+```
+
+`let` `n1` `=` `{ip:` `"127.0.0.1",` `port:` `"8080"};`
+
+`let` `n2` `=`  `{ip:` `"127.0.0.1",` `port:` `"8081"};`
+```
+
```
Comment (vagos) [inline docs/README.md:70]: Remove extra newlines from code snippets :)

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+
+
+
+## FAQ
+
+****
+
+Here we compiled a list of FAQs.
+
```
Comment (vagos) [inline docs/README.md:342]: Don't need this phrase.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+- General Roadmap (for getting started):
+
+    1. Complete `local.js` by adding `status` and `routes`, then `comm `
+    (this is where the initialization of status, routes and comm happens)
+
+    2. Complete `start` in `node.js`
+
+    3. Complete `createRPC` in `wire.js`
+
```
Comment (vagos) [inline docs/README.md:352]: Maybe convert this into an "Implementation Checklist" similar to other milestone handouts.

**Review by vagos — CHANGES_REQUESTED** [review]

**Comment by Scones1016** [comment]
Done!

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,385 @@
+M2: Actors & Remote Procedure Calls
+
+```
+Collaboration: Individual milestone
+Completion: About 10–12 hours
+Deadline: Monday Feb. 19, 2024 (11:59PM ET), pending results of quiz 6
+Latest handout version: CS1380:2024:M2
+GitHub repo: https://github.com/brown-cs1380/m2
+```
+
```
Comment (vagos) [inline docs/README.md:10]: Make this like

>Collaboration: ...
>...
>...

This is done with ">" at the start of the line.

Instead of a code block

**Review by vagos — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 4 above)
Comment (vagos) [reply docs/README.md:10]: Sorry for the double review. Noticed it after. Everything is perfect otherwise :)

Comment (nvasilakis) [reply docs/README.md:10]: +1

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+[Background & Context](#background--context)
```
Comment (nvasilakis) [inline docs/README.md:12]: No need to leave empty lines, can also use bullet points `*`

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+## Per-Node Listener and Configuration
+
+Each node combines two elements: (1) a listening server for receiving messages from the network, explained in this section; and (2) an extensible library of services, shown in the next section, each responsible for handling different messages.
+
+
+
+**Node configuration.** At a minimum, the node configuration includes an IP address and a port. In the examples below, we assume two nodes — by default a node `n1`, which at times initiates interactions with a node `n2`. 
+
+```
+`let` `n1` `=` `{ip:` `"127.0.0.1",` `port:` `"8080"};`
```
Comment (nvasilakis) [inline docs/README.md:63]: You're using both triple backticks _and_ single backticks, which doesn't seem right — use only triple backticks for codeblocks, and use `js` at the end of the top backtick to signal that the code block should be rendered as JavaScript.

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+
+
+
+## Reflections & The Way Forward
+
+As part of your submission add a small README.md markdown file with the following structure, along with answers to the following prompts:
+
+```
+
+\# M2: Actors and Remote Procedure Calls (RPC)
```
Comment (nvasilakis) [inline docs/README.md:264]: Please fix all these backslashed `\` — they're not necessary!

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+## FAQ
+
+- Where to store the node states (route mappings, ids, message counts, etc)?
+
+    - You need to define an object under the `global`, for example `global.myStates = {...}`, then access that object from any js file in your project. 
+
+
+
+- How to replace '\_\_METHOD\_ID\_\_' into the stub?
```
Comment (nvasilakis) [inline docs/README.md:331]: I don't think you need these backslashes `\`.

**Review by nvasilakis — CHANGES_REQUESTED** [review]

**Comment by Scones1016** [comment]
Fixed!

File: docs/README.md [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+**Node configuration.** At a minimum, the node configuration includes an IP address and a port. In the examples below, we assume two nodes — by default a node `n1`, which at times initiates interactions with a node `n2`. 
+
+```js
+let n1 = {ip: "127.0.0.1", port:"8080"};
+let n2 = {ip: "127.0.0.1", port:"8081"};
+```
+
+A node can be configured upon startup either by providing as a parameter a serialized configuration object (in the general case, this object can include any [M1-supported ](https://docs.google.com/document/d/e/2PACX-1vSwf1NRh7D1iYxlqgnIPQT7pvCZEzbwoZ3dsRTnktxzJopuJSU2E2HB7_He-cVcOaDqWgTyYT18gjBJ/pub)deserialization value etc.) or by setting a global `config` object before importing the distribution library. The former is useful for passing a parameter when a node is spawned as a Unix process by running `./distribution.js` in the terminal; the latter is useful for when a node is launched by importing the `distribution.js` library. Here's an example of the former:
+
+```js
```
Comment (nvasilakis) [inline docs/README.md:57]: This snippet is actually shell, use `sh` instead of `js`

File: docs/README.md [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+
+## The Core Service Library
+
+The table below shows a core set of services that need to be implemented (and eventually be made available by the system to anyone importing or using `distribution.js`.):
+
+
+
+|          |                                              |              |
+| -------- | -------------------------------------------- | ------------ |
+| Service  | Description                                  | Methods      |
```
Comment (nvasilakis) [inline docs/README.md:109]: This is the table title, should go above the line!

**Review by nvasilakis — CHANGES_REQUESTED** [review]
Looks awesome! Two minor fixes missed earlier and we're ready to merge!

**Comment by Scones1016** [comment]
Updated

**Review by nvasilakis — APPROVED** [review]

**Pull request merged**